pkg/oc/cli/cmd/login/loginoptions.go

```diff
@@ -104,6 +104,13 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 	// ensure clientConfig has timeout option
 	if o.RequestTimeout > 0 {
 		clientConfig.Timeout = o.RequestTimeout
+
+		// if a timeout was specified, set a custom transport dial func that mimics the default one
+		// from k8s.io/client-go/transport/cache.go#get but with a custom timeout field.
+		clientConfig.Dial = (&net.Dialer{
+			Timeout:   o.RequestTimeout,
```

Comment:
And another comment saying that this is for the http round tripper.

Reply:
So, I took a look at https://github.com/kubernetes/client-go/blob/master/transport/cache.go#L79 which is where the dialer.Dial func ends up being used. Apparently it defaults to the http package's DefaultTransport, [which makes use of a dialer with a 30 second timeout](https://github.com/golang/go/blob/master/src/net/http/transport.go#L42)